### PR TITLE
Allow to specify the pattern for unused bits

### PR DESCRIPTION
**Pull request by andlaus**
The purpose of this is to avoid undefined behavior when encoding frames that contain bits which are not used by any signal. Currently, this pattern implicitly is 0xff if encoding with the `padding` argument set to `True`, and 0x00 otherwise. ARXML allows to specify something arbitrary, so let's also support this in cantools. IMO, this is quite a fringe feature and I'm not even completely sure if it is implemented fully correctly here: I assumed that a single byte gets specified as an integer (thus, endianness does not matter) and that the bit pattern is not aligned with the beginning of unspecified sections, i.e., the encoded message starts out as a sequence of the unspecified bit pattern bytes and then the parts which define signals get overwritten by actual data.

Andreas Lauser &lt;<andreas.lauser@daimler.com>&gt;, on behalf of [MBition GmbH](https://mbition.io/).
[Provider Information](https://github.com/Daimler/daimler-foss/blob/master/PROVIDER_INFORMATION.md)

### CODE REVIEW

**Comment by juleq** [comment]
Do you need to rebase to get the linter satisfied?

**Comment by andlaus** [comment]
> Are you sure that Vector uses 0xff as a default? Will the behavior change when you do not enable padding?

if you set the padding argument of `encode()` to true, cantools will currently set all unused bits  to 1 if it is false, they are set to 0 (i.e. not touched by the codecs). If nothing else gets specified, this should hopefully stay the identical after the patch. I have no idea if the current behavior is what Vector does, though...

**Comment by andlaus** [comment]
> Do you need to rebase to get the linter satisfied?

probably. I'll have a look...

File: cantools/database/can/message.py [hunk 1]
```diff
@@ -65,6 +66,7 @@ def __init__(self,
         self._length = length
         self._signals = signals
         self._signals.sort(key=start_bit)
+        self.unused_bit_pattern = unused_bit_pattern
```
Comment (zariiii9003) [inline cantools/database/can/message.py:69]:
```suggestion
        self._unused_bit_pattern = unused_bit_pattern
```

Comment (andlaus) [reply cantools/database/can/message.py]: oops. fixed, thanks :)

**Comment by andlaus** [comment]
yay! the linter is happy now :)

**Pull request merged**